File: LeetCode/14_Longest_Common_Prefix.py
'''
Write a function to find the longest common prefix string amongst an array of strings.

Input: strs = ["flower","flow","flight"]
Output: "fl"

Input: strs = ["dog","racecar","car"]
Output: ""
'''

def longestCommonPrefix(strs):
    strs.sort()
    z = len(strs)
    min_len = min(len(strs[0]), len(strs[z-1]))

    l = 0
    while l < min_len and strs[0][l] == strs[z-1][l]:
        l += 1
    prefix = strs[0][:l]
    print(prefix)

    # print(strs)
    # print(min_len)






strs = ["flower","flow","flight"]
# strs = ["dog","racecar","car"]

longestCommonPrefix(strs)

File: LeetCode/14_Longest_Common_Prefix.js
var longestCommonPrefix = function(s) {
    s.sort();
    z = s.length;
    min_l = Math.min(s[0].length, s[z-1].length)
    // console.log(s)
    // console.log(min_l)
    left = 0
    while (left < min_l && s[0][left] == s[z-1][left]){
        left += 1
    }
    prefix = s[0].slice(0, left);
    console.log(prefix)




}

s = ["flower","flow","flight"]
// s = ["dog","racecar","car"]
longestCommonPrefix(s)